pkg/services/provisioning/dashboards/file_reader.go

```diff
@@ -234,6 +237,8 @@ func getOrCreateFolderId(cfg *DashboardsAsConfig, service dashboards.DashboardPr
 		dash.Dashboard.IsFolder = true
 		dash.Overwrite = true
 		dash.OrgId = cfg.OrgId
+		// set dashboard folderUid if given
+		dash.Dashboard.SetUid(cfg.FolderUid)
```

Comment:
Seems like this will mean the Uid will not be updatable. This is bit different to how other dashboards work, where you can update them. Basically if anybody wanted to start using it they would have to delete the old folders and reprovision or change the title to seem like a new folder. cc @bergquist is this behaviour ok? Otherwise it seems it would need a bit of refactoring of how we handle the folders as that is how folders work so far.

Reply:
Ah I see it is also mentioned in the description. I assume it is ok now and will need to remember that and maybe refactor later. Especially considering https://github.com/grafana/grafana/pull/16579 which adds reloading API for the config files.

Reply:
I wanted to add this originally, but I agree some refactoring is necessary since changing many of the other fields break the DB.  This decision is up to you guys, but I agree, perhaps belongs in a separate refactoring change since this is not explicitly changing that behavior.